doc/dev/cephadm/design/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,317 @@
+
+A. Current workflow:
+--------------------
+**CLI**:
+    Operations:
+
+.. prompt:: bash #
+
+  ceph orch device ls
+  ceph orch device ls json ( to get all the fields for each device )
+
+    Problems in current implementation:
+        * Does not scale.
+
+**GUI**:
+    Operations:
+        * cluster.Inventory section:
+            The cluster.Inventory section presents a basic list of the devices in the cluster. It is a fixed list with only a few fields. Only the “ident light on” operation is possible although we do not know if it is possible or not until the operation is launched.
+
+    Problems in current implementation:
+        * Does not scale (depends of the orchestrator)
+        * Rigid user experience
+
+B. Proposed workflow:
+---------------------
+
+**CLI**:
+The current API is good enough, we only need to be sure that we have:
+
+    - all the attribute/health/operative state fields from each device
+    - fast response
+    - scalable
+
+**GUI**:
+The inventory should be able to be customized in order to show the desired fields of information for each device. Being customizable also the position of each field(column) and the sort order.
+
+The inventory should be filtered using any of the fields present in the list of devices.
+
+A customized inventory list together with the filter and sort order should be able to be stored for easy utilization. In this way we can provide a set of interesting predefined inventory lists. For example:
+
+    - Devices available
+    - Devices more used (more average iops) (should be an alert/trigger)
+    - Devices biggers than n Gb
+
+The inventory should also provide a way to do directly operations over physical devices:
+
+    * Identify:  Start/stop to blink the identification light
+    * Create OSD: Create an OSD using this disk device if it is available.
+    * Remove OSD: Delete the OSD using this disk device.
+
+2. Add OSDs
+===========
+
+A. Current workflow
+--------------------
+
+**CLI**:
+We can specify specific devices or use a “drive group” specification to create OSD’s in different hosts. By default, the definition of the OSDs to create is “declarative“ unless you use the unmanaged parameter.
+
+.. prompt:: bash #
+
+  ceph orch daemon add osd <host>:device1,device2 [--unmanaged=true]  (manual approach)
+  ceph orch apply osd -i <json_file/yaml_file> [--dry-run] [--unmanaged=true]*  (Service Spec based approach)
+
+**GUI**:
+Implemented in the dashboard section “cluster.OSDs”.
+There is a button to create the OSDs, that presents a page dialog box to select the physical devices that are going to be part of the OSDs as main, WAL or Db devices.
+It Is very difficult to make a selection ( or to understand how to make the selection). This is even worse if your cluster has the same kind of devices, resulting in the weird thing that is not possible to create an OSD using only one storage device (because you cannot select it)
```

Comment:
```suggestion
It is very difficult to make a selection (or to understand how to make the selection). This is even worse if your cluster has the same kind of devices, resulting in the weird thing that is not possible to create an OSD using only one storage device (because you cannot select it)
```

Reply:
I know that this document is in an early stage of its development and that at present it represents a mere list of ideas, but I suggest that this paragraph, in the interest of reducing the cognitive load of the reader, be changed to read as follows:

"It is difficult to make a selection and difficult to determine how to make a selection. This is especially true if your cluster consists of devices of the same kind. If your cluster is like this, you will be in the unusual case of begin unable to create an OSD using only one storage device, because you will not be able to select it."